benchmark_runner/common/template_operations/template_operations.py

```diff
@@ -94,12 +94,15 @@ def __build_template_data(self, previous_data: dict, template_root_data: dict):
                 **extra_data, **extra_kind_data, **extra_runtype_data, **extra_kind_runtype_data}
 
     @logger_time_stamp
-    def generate_yamls_internal(self, scale: str = None, scale_node: str = None, redis: str = None):
+    def __generate_yamls_internal(self, scale: str = None, scale_num: str = None, scale_node: str = None, redis: str = None):
```

Comment:
I still don't like casting it outside, but marking this function private guarantees that it is not an API that somebody else can use.